### ISSUE TITLE
Missing typescript declaration file for module

### ISSUE THREAD

**Issue by niravbhimani53** [roderickhsiao/react-in-viewport#47]
Could not find a declaration file for module 'react-in-viewport'. '/Users/Projects/xyz/node_modules/react-in-viewport/dist/index.js' implicitly has an 'any' type.

Try `npm install @types/react-in-viewport` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-in-viewport';`ts(7016)

I'm using typescript for my project and tsconfig.json got strict mode true.

Would you like me to make one in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped)? Or is it already done?

**Comment by roderickhsiao** [comment]
Absolutely! I’m not too familiar with typescript so if you could help that
would be awesome!

**Comment by GuiMercier1** [comment]
Hello, "a minima" I've solved it by creating the following "react-in-viewport.d.ts" file in src/@types :

```
declare module 'react-in-viewport' {
    function handleViewport(block: ReactNode, options: any, config: any): any
    export = handleViewport
}
```

Obviously it must be improved later to describe props, options and config, but at least your lint won't block you.

**Comment by roderickhsiao** [comment]
@GuiMercier1 Appreciated!

**Comment by neolefty** [comment]
Yes, types would be super handy. I'd like to be able to use `useInViewport` in a Typescript project.

**Comment by timur-mut** [comment]
```
declare module 'react-in-viewport' {
  import {ReactNode} from "react";
  import {MutableRefObject} from "react";
  export declare function useInViewport(ref: MutableRefObject<any>, options?: any, config?: any, props: any): {
    inViewport: boolean,
    enterCount: number,
    leaveCount: number,
  }

  function handleViewport(block: ReactNode, options?: any, config?: any): any
  export = handleViewport
}

```

**Comment by roderickhsiao** [comment]
fixed in 1.0.0-alpha.17 #87
